apps/src/netsim/NetSimConnection.js

```diff
@@ -94,7 +94,10 @@ var NetSimConnection = module.exports = function (sentLog, receivedLog) {
   this.statusChanges = new ObservableEvent();
 
   // Bind to onBeforeUnload event to attempt graceful disconnect
-  window.addEventListener('beforeunload', this.onBeforeUnload_.bind(this));
+  // Null-guard so constructing this object is test-friendly.
+  if (window && window.addEventListener) {
+    window.addEventListener('beforeunload', this.onBeforeUnload_.bind(this));
+  }
```

Comment:
I was adding an empty addEventListener method to window in tests for a bit, but linting complained about the window object being readonly.  I'll change this up so we have to pass window into NetSimConnection; that way we can construct a fake.